metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/dao/ElasticsearchMetaAlertDao.java

```diff
@@ -295,19 +341,199 @@ protected Document buildCreateDocument(MultiGetResponse multiGetResponse, List<S
   /**
    * Process an update to a meta alert itself.
    * @param update The update Document to be applied
-   * @param index The optional index to update to
    * @throws IOException If there's a problem running the update
    */
-  protected void handleMetaUpdate(Document update, Optional<String> index) throws IOException {
-    // We have an update to a meta alert itself
-    // If we've updated the alerts field (i.e add/remove), recalculate meta alert scores.
+  protected void handleMetaUpdate(Document update) throws IOException {
```

Comment:
If I have a metaalert with a large number of alerts (say 525 in this example), then I attempt to add 1 or 2 alerts to that metaalert, the REST API responds back with a 500 error.  

I don't see any error messages in the logs.  I assume that error is coming from somewhere in here.

The UI itself never reports an error or acknowledges.  Of course that is a separate problem that we should fix in a separate PR.


![screen shot 2017-10-24 at 6 53 24 pm](https://user-images.githubusercontent.com/2475409/31972179-5e89d79c-b8ed-11e7-9e7a-7ac6c0b718c8.png)

But I do find that the metaalert was actually updated, as the number of alerts increased (from 527 to 529 here).

![screen shot 2017-10-24 at 6 59 04 pm](https://user-images.githubusercontent.com/2475409/31972201-78e06b24-b8ed-11e7-80c5-4bb01a3e72b7.png)

Reply:
I'll try to reproduce. I'm not sure what the potential cause could be.  Could you see if the alerts you added have the properly populated "metaalerts" field in the original alert?  It's possible an issue occurs during that update, but I don't really have any evidence in any direction.

Reply:
I can help debug further.  I just wanted to get the report of the problem out there first.  I'll research how to kick-up logging on the API so we can actually see what's going on.

Reply:
Here is the exception that pops out with that 500.

```
"fullMessage":"IOException: class org.apache.metron.indexing.dao.HBaseDao: KeyValue size too large
java.lang.IllegalArgumentException: KeyValue size too large
\tat org.apache.hadoop.hbase.client.HTable.validatePut(HTable.java:1521)
\tat org.apache.hadoop.hbase.client.BufferedMutatorImpl.validatePut(BufferedMutatorImpl.java:147)
\tat org.apache.hadoop.hbase.client.BufferedMutatorImpl.doMutate(BufferedMutatorImpl.java:134)
\tat org.apache.hadoop.hbase.client.BufferedMutatorImpl.mutate(BufferedMutatorImpl.java:105)
\tat org.apache.hadoop.hbase.client.HTable.put(HTable.java:1050)
\tat org.apache.metron.indexing.dao.HBaseDao.batchUpdate(HBaseDao.java:140)
\tat org.apache.metron.indexing.dao.MultiIndexDao.lambda$batchUpdate$3(MultiIndexDao.java:78)
\tat java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
\tat java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
\tat java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
\tat java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
\tat java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:747)
\tat java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:721)
\tat java.util.stream.AbstractTask.compute(AbstractTask.java:316)
\tat java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
\tat java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
\tat java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:401)
\tat java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:734)
\tat java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:714)
\tat java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
\tat java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
\tat org.apache.metron.indexing.dao.MultiIndexDao.batchUpdate(MultiIndexDao.java:83)
\tat org.apache.metron.elasticsearch.dao.ElasticsearchMetaAlertDao.indexDaoUpdate(ElasticsearchMetaAlertDao.java:374)
\tat org.apache.metron.elasticsearch.dao.ElasticsearchMetaAlertDao.handleMetaUpdate(ElasticsearchMetaAlertDao.java:361)
\tat org.apache.metron.elasticsearch.dao.ElasticsearchMetaAlertDao.update(ElasticsearchMetaAlertDao.java:248)
\tat org.apache.metron.indexing.dao.IndexDao.patch(IndexDao.java:130)
\tat org.apache.metron.rest.service.impl.UpdateServiceImpl.patch(UpdateServiceImpl.java:44)
\tat org.apache.metron.rest.controller.UpdateController.patch(UpdateController.java:52)
```

Reply:
The problem here seems unrelated to your PR.  I think the serialized JSON is just too large with these ~500 alerts when it attempts to write them to HBase.  I think we impose a limit of 999 currently, but I'm not sure a fixed limit is going to always work.  It depends on the size of each alert.

I will create a JIRA to track this separately.

Reply:
https://issues.apache.org/jira/browse/METRON-1279

Reply:
I added a comment on https://github.com/apache/metron/pull/803/, because the 999 limit is actually only established there, it's not something in master.